src/app/clusters/operational-state-server/operational-state-delegate.h

```diff
@@ -0,0 +1,308 @@
+     */
+    virtual void GetOperationalState(GenericOperationalState & op) = 0;
+
+    /**
+     * Get the list of supported operational states.
+     * Fills in the provided GenericOperationalState with the state at index `index` if there is one,
+     * or returns CHIP_ERROR_NOT_FOUND if the index is out of range for the list of states.
+     * @param index The state of index starts at 0.
+     * @param operationalState  The GenericOperationalState is filled.
+     */
+    virtual CHIP_ERROR GetOperationalStateAtIndex(size_t index, GenericOperationalState & operationalState) = 0;
+
+    /**
+     * Get the list of supported operational phase.
+     * Fills in the provided GenericOperationalPhase with the state at index `index` if there is one,
+     * or returns CHIP_ERROR_NOT_FOUND if the index is out of range for the list of states.
+     * @param index The state of index starts at 0.
+     * @param operationalPhase  The GenericOperationalPhase is filled.
+     */
+    virtual CHIP_ERROR GetOperationalPhaseAtIndex(size_t index, GenericOperationalPhase & operationalPhase) = 0;
+
+    /**
+     * Get current operational error.
+     * @param error.Put a struct instance on the state, then call the delegate to fill it in.
+     */
+    virtual void GetOperationalError(GenericOperationalError & error) = 0;
+
+    /**
+     * Set operational state.
+     * @param opState The operational state for which to set.
+     * @return CHIP_ERROR CHIP_NO_ERROR on success, or corresponding error code.
+     */
+    virtual CHIP_ERROR SetOperationalState(const GenericOperationalState & opState) = 0;
+
+    /**
+     * Set operational error.
+     * @param opErrState The reference of operational error.
+     * @return CHIP_ERROR CHIP_NO_ERROR on success, or corresponding error code.
+     */
+    virtual CHIP_ERROR SetOperationalError(const GenericOperationalError & opErrState) = 0;
+
+    // command callback
+    /**
+     * Handle Command Callback in application: Pause
+     * @param[out] get operational error after callback.
+     */
+    virtual void HandlePauseStateCallback(GenericOperationalError & err) = 0;
+
+    /**
+     * Handle Command Callback in application: Resume
+     * @param[out] get operational error after callback.
+     */
+    virtual void HandleResumeStateCallback(GenericOperationalError & err) = 0;
+
+    /**
+     * Handle Command Callback in application: Start
+     * @param[out] get operational error after callback.
+     */
+    virtual void HandleStartStateCallback(GenericOperationalError & err) = 0;
+
+    /**
+     * Handle Command Callback in application: Stop
+     * @param[out] get operational error after callback.
+     */
+    virtual void HandleStopStateCallback(GenericOperationalError & err) = 0;
+
+    Delegate(EndpointId aEndpointId, ClusterId aClusterId) : mEndpointId(aEndpointId), mClusterId(aClusterId) {}
+
+    virtual ~Delegate() = default;
+
+protected:
+    EndpointId mEndpointId;
+    ClusterId mClusterId;
```

Comment:
Where are these used?  I don't see them being used.  They should probably just be removed....